File: problems/Two Pointers/DuplicateNumbersII.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        l = 0
        r = 1
        i = 1

        if n < 3:
            return n
        counter = 0
        while r < n:
            if(nums[l] != nums[r]):
                l += 1
                nums[l] = nums[r]
                counter = 0
            elif(nums[l] == nums[r] and counter < 1):
                counter += 1
                l += 1
                nums[l] = nums[r]
            r += 1
        return l+1

File: problems/Java/DuplicateNumbersII.java
class Solution {
	public int removeDuplicates(int[] nums) {
		int n = nums.length;
		int l = 0;
		int r = 1;
		int counter = 0;

		while (r < n) {
			if (nums[l] == nums[r] && counter < 1) {
				l++;
				nums[l] = nums[r];
				counter++;
			} else if (nums[l] != nums[r]) {
				l++;
				nums[l] = nums[r];
				counter = 0;

			}
			r++;
		}
		return l + 1;
	}
}